gslab_scons/tests/test_configuration_tests.py

```diff
@@ -0,0 +1,394 @@
+        # But now yaml has a correct key and doesn't require user input.
+        mock_raw_input.side_effect = lambda x: "bad_value"
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+        mock_raw_input.side_effect = raw_input_side_effect
+
+        # Yaml file exists and is corrupt. User deletes and re-creates. 
+        make_yaml('key value')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters value to create. 
+        os.remove('yaml.yaml')
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'key'), 'value')
+
+        # Yaml file does not exist. User enters none value to create. 
+        os.remove('yaml.yaml')
+        mock_raw_input.side_effect = raw_input_side_effect2
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Yaml file now exists with none value.
+        self.assertEqual(configuration_tests.load_yaml_value('yaml.yaml', 'stata_executable'), None)
+
+        # Test bad
+        with self.assertRaises(ex_classes.PrerequisiteError):
+            # Corrupt file and user doesn't choose to delete and re-create 
+            make_yaml('key value')
+            configuration_tests.load_yaml_value('yaml.yaml', 'key')
+
+        if os.path.isfile('yaml.yaml'):
+            os.remove('yaml.yaml')  
+
+
+    @mock.patch('gslab_scons.configuration_tests.is_unix')
+    @mock.patch('gslab_scons.configuration_tests.convert_packages_argument')
+    @mock.patch('gslab_scons.configuration_tests.subprocess.check_output')
+    def test_check_stata_packages(self, mock_check_output, mock_package_argument, 
+                                  mock_is_unix):
+
+        def check_output_side_effect(*args, **kwargs):
+            command = args[0].split("which ")[0]
+            if not re.search("statamp", command):
+                raise subprocess.CalledProcessError(1, args[0])  
+
+            with open('stata.log', 'wb') as f:
+                if not re.search('yaml', args[0]):
+                    package = args[0].split("which ")[1]
+                    f.write("command %s not found" % package)
+
+        def is_unix_side_effect_true():
+            return True
+ 
+        def is_unix_side_effect_false():
+            return False
+
+        mock_check_output.side_effect     = check_output_side_effect
+        mock_package_argument.side_effect = lambda x: ['yaml']
+        mock_is_unix.side_effect          = is_unix_side_effect_true
+
+        with mock.patch('gslab_scons.configuration_tests.sys.platform', 'win32'):
+            # Good tests
+
+            # Unix
+            configuration_tests.check_stata_packages("statamp", "yaml")
+
+            # Windows
+            mock_is_unix.side_effect = is_unix_side_effect_false
+            configuration_tests.check_stata_packages("statamp", "yaml")
+
+            # No packages
```

Comment:
Maybe add assertions to confirm that the function is making the check we expect it to?